crates/ruma-common/src/push/condition.rs

```diff
@@ -456,7 +456,7 @@ mod tests {
         assert!("foobar".matches_pattern("foo*", false));
         assert!("foo bar".matches_pattern("foo*", false));
         assert!(!"foo".matches_pattern("foo?", false));
-        assert!("foo".matches_pattern("fo?", false));
+        assert!("fooo".matches_pattern("foo?", false));
```

Comment:
This change is because `fo?` triggers a false positive for `of` or `for` and I feel like it would be a common error so I didn't want to disable it.